### PR TITLE
Time-delay likelihood correction for multiple point sources

### PR DESCRIPTION
**Pull request by martin-millon**
Solve issue #678. 

Changes : 
- Allows to provide list of list of booleans for each point source and for each image to determine which time-delays are used
- Propagation of the z_lens_convention argument in multiplane 
- Introduce the color bar option in the convergence plot 
- Propagate the lens equation solver options to the point source plots

### CODE REVIEW

File: lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py [hunk 1]
```diff
@@ -23,7 +23,9 @@ def __init__(
         :param lens_model_class: instance of the LensModel() class
         :param point_source_class: instance of the PointSource() class, note: the first point source type is the one the
          time delays are imposed on
-        :param time_delay_measurement_bool_list: list of bool to indicate for which point source model a measurement is available
+        :param time_delay_measurement_bool_list: list of list of bool to indicate for which point source model a measurement is available.
+        This list must have the same length as time_delays_measured and time_delays_uncertainties.
```
Comment (sibirrer) [inline lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py:27]: for the second line, there should be a space inwards to be rendered properly

File: lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py [hunk 2]
```diff
@@ -49,7 +51,23 @@ def __init__(
...
+                for i in range(self._num_point_sources):
+                    time_delay_measurement_bool_list.append(
+                        [True] * len(time_delays_measured[i])
+                    )
+        else:
+            for i in range(self._num_point_sources):
+                if len(time_delay_measurement_bool_list[i]) != len(
+                    self._delays_measured[i]
+                ):
+                    raise ValueError(
```
Comment (sibirrer) [inline lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py:67]: this error does not seem to be backwards compatible. A single True per point source model should also be possible and then resulting in the delays being calculated and evaluated between all of the point sources

File: lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py [hunk 3]
```diff
@@ -76,9 +95,17 @@ def logL(self, kwargs_lens, kwargs_ps, kwargs_cosmo):
...
+                mask_full = np.concatenate(
+                    ([True], mask)
+                )  # add the first image to the mask
+                if len(delay_days) - 1 != len(self._delays_measured[i]):
+                    logL += -(10**15)
+                else:
+                    logL += self._logL_delays(
+                        delay_days[mask_full],
+                        self._delays_measured[i][mask],
+                        self._delays_errors[i][mask],
```
Comment (sibirrer) [inline lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py:107]: i am not sure this works when providing a n x n covariance matrix as the errors in the delays

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you @martin-millon ! I have some comments and questions whether the changes will work with error covariance matrices. I'd still like to have a single boolean per point source model be supported, if possible. 
There is also some text coverage dropping due to mostly small changes in the plotting routine.

**Comment by martin-millon** [comment]
@sibirrer I am still working on this PR, especially the automated test, which needs to be updated. I'll let you know when it's ready and include your comments in the new version.

File: lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py (hunk 3 above)
Comment (martin-millon) [reply lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py]: This is now tested in L220 of test_time_delay_likelihood.py and it works

File: lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py (hunk 2 above)
Comment (martin-millon) [reply lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py:85]: It should be backward compatible now

File: lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py (hunk 1 above)
Comment (martin-millon) [reply lenstronomy/Sampling/Likelihoods/time_delay_likelihood.py]: corrected

**Comment by martin-millon** [comment]
@sibirrer I think this is now ready to be merged.

File: lenstronomy/Plots/lens_plot.py [hunk 4]
```diff
@@ -161,14 +164,15 @@ def convergence_plot(
     :param vmax: matplotlib vmax
     :param cmap: matplotlib cmap
     :param kwargs: keyword arguments for matshow
+    :param with_color_bar: bool, if True, shows color bar
```
Comment (sibirrer) [inline lenstronomy/Plots/lens_plot.py:167]: can you add docstrings for colorer_label and font_size?

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks good. Just some minor docstrings to be added and I am waiting for the coverage report

**Comment by martin-millon** [comment]
I don't get why these two lines of code do not get executed by codecov whereas the program enters the if statement...

**Pull request merged**